packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -678,6 +678,24 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
     this.validate();
   }
 
+  __stepChanged(step) {
+    if (step === undefined) {
+      return;
+    }
+
+    const parsedObj = this.i18n.parseTime(this.i18n.formatTime(TEST_TIME_OBJ));
+    if (
+      (step % 1 !== 0 && !parsedObj.milliseconds) ||
+      (step % 60 !== 0 && !parsedObj.seconds) ||
+      (step % (60 * 60) !== 0 && !parsedObj.minutes)
+    ) {
+      console.warn(
+        `<vaadin-time-picker> The step ${step} seconds is not compatible with the provided time formatter, so it has been unset.`,
+      );
+      this.step = undefined;
```

Comment:
Should we restore old `step` value, if any?

Reply:
We could do that. 👍

Reply:
Done.